exetera/core/dataframe.py

```diff
@@ -1600,12 +1595,12 @@ def _ordered_merge(left: DataFrame,
                     a_on[0], b_on[0], a_result, b_result, invalid, rdtype=npdtype)
 
         if how == 'right':
-            dest.rename('_a_map', '_right_map')
+            dest.rename('_a_map', '_right_map') if '_a_map' in dest else None
```

Comment:
The `if-else` expression should be used where an expression is expected only and not as a substitute for an `if` statement, so this would be 

```python
if "_a_map" in dest:
    dest.rename('_a_map', '_right_map')
```